app/src/main/java/swati4star/createpdf/fragment/MergeFilesFragment.java

```diff
@@ -0,0 +1,311 @@
+                noSupport();
+            } else {
+                mergeBtn.setEnabled(true);
+                mergePdf(srcs);
+                Toast.makeText(this.getContext(), getString(R.string.pdf_merge), Toast.LENGTH_SHORT).show();
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void mergePdf(String[] srcs) {
+        try {
+            // Create document object
+            Document document = new Document();
+            // Create pdf copy object to copy current document to the output mergedresult file
+            mPath = Environment.getExternalStorageDirectory().getAbsolutePath() +
+                    MergeFilesFragment.this.getString(R.string.pdf_dir);
+            mFilename = mFilename + getString(R.string.pdf_ext);
+            String finPath = mPath + mFilename;
+            PdfCopy copy = new PdfCopy(document, new FileOutputStream(finPath));
+            // Open the document
+            document.open();
+            PdfReader pr;
+            int n;
+            for (int i = 0; i < srcs.length; i++) {
+                // Create pdf reader object to read each input pdf file
+                pr = new PdfReader(srcs[i]);
+                // Get the number of pages of the pdf file
+                n = pr.getNumberOfPages();
+                for (int page = 1; page <= n; page++) {
+                    // Import all pages from the file to PdfCopy
+                    copy.addPage( copy.getImportedPage(pr , page));
+                }
+            }
+            document.close(); // close the document
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void showFileChooser() {
+
+        String folderPath = Environment.getExternalStorageDirectory() + "/";
+        Intent intent = new Intent();
+        intent.setAction( Intent.ACTION_GET_CONTENT);
+        Uri myUri = Uri.parse(folderPath);
+        intent.setDataAndType( myUri , getString(R.string.pdf_type));
+        Intent intentChooser = Intent.createChooser(intent , getString(R.string.merge_file_select));
+        startActivityForResult(intentChooser , INTENT_REQUEST_PICKFILE_CODE);
+    }
+
+
+    public void onActivityResult(int requestCode, int resultCode, Intent data) throws NullPointerException {
+        if (data != null) {
+
+            if (requestCode == INTENT_REQUEST_PICKFILE_CODE ) {
+                if (resultCode == RESULT_OK) {
+                    Uri uri = data.getData();
+                    String uriString = uri.toString();
+                    File myFile = new File(uri.toString());
+                    String path = myFile.getPath();
+                    String realPath = Environment.getExternalStorageDirectory().getAbsolutePath();
+                    String displayName = null;
+                    if (addFileOne.getTag().toString().equals(btTag)) {
+                        if (uriString.startsWith("content://")) {
```

Comment:
You are doing the exact same thing for first as well as the second PDF file chooser.
You can move this piece of code to another function, pass relevant parameters.
Code looks quite complex right now.

Also, add comments.